Controller panic after upgrade to v2.6.0 on Kubernetes 1.35.3: "unable to calculate an index entry for key ... on index 'nodeIPs'"

**Describe the bug**
After upgrading Antrea from v2.5.2 to **v2.6.0**, the `antrea-controller` crashes with a panic in one of my clusters.

The crash happens in the `nodeIPs` index when processing Node objects:
```
panic: unable to calculate an index entry for key "k8s-c-01" on index "nodeIPs": no IP with type in [InternalIP ExternalIP] was found for Node 'k8s-c-01'
```
The panic only started after the upgrade to 2.6.0 and appears to be triggered during cache rebuild / informer processing.
This occurs even though **all nodes have a valid `InternalIP`** in `.status.addresses`.

Additionally, we see repeated non-fatal errors (on both clusters):
```
E... [SHOULD NOT HAPPEN] failed to update managedFields
err="failed to convert new object (/...; controlplane.antrea.io/v1beta2, Kind=NodeStatsSummary) to smd typed: no corresponding type for controlplane.antrea.io/v1beta2, Kind=NodeStatsSummary"
```

**To Reproduce**
1. Run a cluster with Kubernetes 1.34.3, Talos 1.11.6 and Antrea v2.5.2.
2. Upgrade to  Kubernetes 1.35.3, Talos 1.12.6 and Antrea v2.6.0
3. Observe controller logs → panic on `nodeIPs` index for some nodes.
4. The managedFields warning for `NodeStatsSummary` also appears after agents are updated.

**Expected**
Controller should start and run stably without panicking. The `nodeIPs` index should handle nodes that correctly report `InternalIP`.

**Actual behavior**
- Controller repeatedly crashes with the `nodeIPs` panic.
- `NodeStatsSummary` managedFields conversion errors spam the logs.

**Versions:**
Please provide the following information:
 - Antrea version v2.6.0
 - Kubernetes version v1.35.3
 - Container runtime: containerd v2.1.6
 - Linux kernel version: 6.18.18
 - Talos version: v1.12.6

**Additional context**
- All nodes correctly show `InternalIP` when queried with:
  ```bash
  kubectl get nodes -o jsonpath='{range .items[*]}{.metadata.name}{"\t"}{.status.addresses}{"\n"}{end}'

--- a/pkg/controller/grouping/controller.go
+++ b/pkg/controller/grouping/controller.go
@@ -69,9 +69,14 @@ func NodeIPsIndexFunc(obj interface{}) ([]string, error) {
 		return nil, nil
 	}
 
+	// We should not return an error if no IP is found as it can be a transient condition, and
+	// it would cause a panic.
+	// In practice, the only reason for k8s.GetNodeAllAddrs to return an error is if no matching
+	// IP address is found for the Node.
 	ips, err := k8s.GetNodeAllAddrs(node)
 	if err != nil {
-		return nil, err
+		klog.V(3).InfoS("Failed to get addresses for Node", "node", klog.KObj(node), "err", err)
+		return nil, nil
 	}
 	return ips.UnsortedList(), nil
 }

--- a/pkg/controller/grouping/controller_test.go
+++ b/pkg/controller/grouping/controller_test.go
@@ -198,7 +198,7 @@ func TestNodeIPsIndexFunc(t *testing.T) {
 				Spec: v1.NodeSpec{PodCIDR: "10.0.0.0/8"},
 			}},
 			want:    nil,
-			wantErr: true,
+			wantErr: false,
 		},
 		{
 			name:    "empty PODCIDR",
@@ -213,7 +213,21 @@ func TestNodeIPsIndexFunc(t *testing.T) {
 				Spec:   v1.NodeSpec{PodCIDR: "10.0.0.0/8"},
 			}},
 			want:    nil,
-			wantErr: true,
+			wantErr: false,
+		},
+		{
+			name: "invalid IP address",
+			args: args{obj: &v1.Node{
+				Status: v1.NodeStatus{Addresses: []v1.NodeAddress{
+					{
+						Type:    v1.NodeInternalIP,
+						Address: "not-a-valid-ip",
+					},
+				}},
+				Spec: v1.NodeSpec{PodCIDR: "10.0.0.0/8"},
+			}},
+			want:    nil,
+			wantErr: false,
 		},
 		{
 			name: "Node with IPs",

